lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,28 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+    # converts the include hash to a standard format
+    # for constrained serialization recursion
+    #
+    # converts
+    #  [:author, :comments => [:author]] to
+    #  {:author => {}, :comments => { author: {} } }
+    #
+    # and
+    #  [:author, :comments => {:author => :bio}, :posts => [:comments]] to
+    #  {:author => {}, :comments => { author: { bio: {} } }, :posts => { comments: {} } }
+    #
+    # The data passed in to this method could be an array, hash or symbol
+    #
+    # the point of this method is to normalize the include
+    # options for the child relationships.
+    def include_options_to_hash(included)
```

Comment:
`include_args_to_hash`?  not really options, which is also the word we use for hashes...

Also, syntax and questions

```ruby
  case included
  when Symbol
    then { included => {} }
  when Hash
    included.reduce({}) { |hash, (k, v)| hash[k] = include_args_to_hash(v) } # no reason to convert to array then to hash.. straight to hash!
  when Array 
    included.reduce({}) {|hash, element| hash.merge!(include_args_to_hash(element) } # merge! creates less objects.. just mutates the one
  else  fail ArgumentError, "Cannot convert #{included.class},  only Symbol, Hash, Array: #{included.inspect}"
  end
end
```